subprojects/plugins/src/main/java/org/gradle/api/plugins/WarPlugin.java

```diff
@@ -112,6 +117,8 @@ public void configureConfigurations(ConfigurationContainer configurationContaine
     }
 
     private void configureComponent(Project project, PublishArtifact warArtifact) {
-        project.getComponents().add(objectFactory.newInstance(WebApplication.class, warArtifact, "master"));
+        AttributeContainer attributes = attributesFactory.mutable()
+            .attribute(Usage.USAGE_ATTRIBUTE, objectFactory.named(Usage.class, Usage.JAVA_RUNTIME));
+        project.getComponents().add(objectFactory.newInstance(WebApplication.class, warArtifact, "master", attributes));
```

Comment:
Nice that we catched and fixed this. :+1:

Chances are that the new validation will reveal other issues with "custom publishing solutions".